test_integration/lib/test/realtime/realtime_subscribe.dart

```diff
@@ -1,26 +1,30 @@
 import 'package:ably_flutter/ably_flutter.dart';
 import 'package:ably_flutter_example/provisioning.dart';
+import 'package:ably_flutter_integration_test/utils/data.dart';
 
 import '../../factory/reporter.dart';
 import '../../utils/encoders.dart';
 import '../../utils/realtime.dart';
 
-Future<List<Map<String, dynamic>>> _getAllMessages(
-  String apiKey,
-  String channelName, {
-  String name,
-  List<String> names,
-}) async {
-  final messages = <Map<String, dynamic>>[];
-
+Future<Realtime> _createRealtime(String apiKey) async {
   final realtime = Realtime(
     options: ClientOptions.fromKey(apiKey)
       ..environment = 'sandbox'
       ..clientId = 'someClientId'
       ..autoConnect = false,
   );
   await realtime.connect();
+  return realtime;
+}
 
+Future<List<Map<String, dynamic>>> _getAllMessages(
+  String apiKey,
+  String channelName, {
+  String name,
+  List<String> names,
```

Comment:
Whats the difference between `channelName`, `name` and `names`? I think the parameter names (and also the arguments used) are a bit confusing. I guess my confusion stems from `channel.subscribe(name: name, names: names)`.

Reply:
Anyhow to avoid confusion, I'm going to rename these arguments as `messageName` and `messageNames`

Reply:
Would a user ever need to use both `messageName` and `messageNames` simultaneously? If not, can you also limit it to `names`, or split them into 2 functions?

In TS, they have union types, so it makes sense in `messageCallback`. However, taking 2 forms of `messageName` simultaneously is a bit complicated.

Reply:
Limiting it to `names` sounds better, but that's against the spec.

Dart doesn't allow overloading: https://dart.dev/guides/language/effective-dart/design#avoid-using-runtime-type-tests-to-fake-overloading

And union types are in consideration: https://github.com/dart-lang/language/issues/1222 & https://github.com/dart-lang/language/issues/83

Reply:
> Limiting it to names sounds better, but that's against the spec.

Do you have the feature spec point that says that? 

> Dart doesn't allow overloading
Yup, but that doesn't mean you can't have 2 functions, and to quote your first link:

> If callers usually know which type they have and which specific operation they want, it’s better to define separate methods with different names to let callers select the right operation. 

They don't mention making one function which takes all (both) the types a user might want to use.

> And union types are in consideration

Yea, I meant you're taking inspiration from typescript's union type, and unfortunately it doesn't work in Dart.

Reply:
I see in the root readme you state:

> Use `channel.subscribe(name: "event1")` or `channel.subscribe(names: ["event1", "event2"])` to listen to specific named messages.

I also note that this means of defining method signature also applies to publish.

Regardless, @ben-xD, I'm not keen to be changing the plugin public API at this stage. Though I would appreciate @tiholic validating that those examples in the readme both work as I do find the presence of multiple arguments on `subscribe` (`name` and `names`) confusing. Especially bearing in mind the API commentary for that method doesn't mention this in any way.

Also, for what it's worth, I don't think that JavaScript interfaces are a good source of inspiration for Dart interfaces... I would have preferred you looked to Java or Cocoa for that. 🤔

Reply:
Java supports both `name` and `names` with help of overloading, but looks like cocoa supports only `name`.

Created an issue to track this: https://github.com/ably/ably-flutter/issues/131